sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/index/SupportsIndex.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.connector.catalog.index;
+
+import java.util.Map;
+import java.util.Properties;
+
+import org.apache.spark.annotation.Evolving;
+import org.apache.spark.sql.catalyst.analysis.IndexAlreadyExistsException;
+import org.apache.spark.sql.catalyst.analysis.NoSuchIndexException;
+import org.apache.spark.sql.catalyst.analysis.NoSuchTableException;
+import org.apache.spark.sql.connector.catalog.CatalogPlugin;
+import org.apache.spark.sql.connector.catalog.Identifier;
+import org.apache.spark.sql.connector.expressions.NamedReference;
+
+/**
+ * Catalog methods for working with index
+ *
+ * @since 3.3.0
+ */
+@Evolving
+public interface SupportsIndex extends CatalogPlugin {
```

Comment:
shall we follow `SupportsPartitionManagement` and make it extends `Table`?

Reply:
nvm, index has a unique name. DROP INDEX does not need a table.

Reply:
Actually, not all databases make index name globally unique, see https://www.w3schools.com/sql/sql_ref_drop_index.asp

I think we can still make `SupportsIndex` extends `Table`, if the SQL syntax is `DROP INDEX index_name ON table_name;`

Reply:
After thinking more about it, I think `DROP INDEX index_name ON [TABLE] table_name` is better, as it's more consistent with the `CREATE INDEX` syntax.

This is also more flexible: the index name only need to be unique within the table.